### DISCUSSION TITLE
Maintaining Settings

### DISCUSSION THREAD

**Discussion by SJLennon** [codefori/vscode-ibmi#349] [Q&A]
I'm confused about settings, which seem to be maintained in different ways/places.

1. From Setting at the bottom of the screen:
![image](https://user-images.githubusercontent.com/67484051/137550101-eaa96d9a-73c8-418c-85d2-a8d85518509f.png)

2. From File/Preferences/Settings 
![image](https://user-images.githubusercontent.com/67484051/137550456-878db13b-059c-49ee-93f2-c9cc106a78f7.png)

3. Editing directly in settings.json.

Maybe  1. is settings that are specific to a connection profile and 2. & 3. are settings that are connected to the VSCODE user?
This is somewhat confounded by the **Help, which I had thought to update and which triggered my confusion**.

For example, this seems to no longer be relevant:
![image](https://user-images.githubusercontent.com/67484051/137551475-015ef1c6-fd19-4785-ab75-1b0c7d15c676.png)

@worksofliam Can you clarify what direction maintaining settings is taking?  (Or enlighten my confusion. 🥲 )

**Comment by worksofliam** [comment]
Yes.. it's a bit confusing and could maybe be fixed with a cleanup.

Screenshot 1 is used for managing Global configurations which affect the extension (and therefore every connection)
Screenshot 2 is specific to settings for the current connection.
Screenshot 3 is just a JSON editor, and the 'Source File list' you see is likely an old outdated setting (from before we had connection settings). If you took it out of the JSON settings, you wouldn't see it there anymore.

Hope that helps. Maybe we need to make this stuff easier. Will think about it.

Liam
